rnd/autogpt_server/autogpt_server/executor/manager.py

```diff
@@ -17,7 +18,8 @@
     ExecutionQueue,
 )
 from autogpt_server.data.graph import Node, get_node, get_graph
-from autogpt_server.util.service import AppService, expose
+from autogpt_server.util.service import AppService, expose, get_service_client  # type: ignore
+from autogpt_server.server import AgentServer
```

Comment:
Here is the issue I was concern about in going with this [tightly coupled](https://github.com/Significant-Gravitas/AutoGPT/pull/7212#pullrequestreview-2117835196) approach using pyro5.

To effectively setup the 2 way communication. The `executor.manager` needs to have imported the `server.AgentServer` to be able to send events to it. 

And the `server.AgentServer` needs to have imported the `executor.manager` to be able to send execution requests to the executors. 

Naturally leading to this:

```
ImportError: cannot import name 'ExecutionManager' from partially initialized module 'autogpt_server.executor' (most likely due to a circular import) (autogpt_server/autogpt_server/executor/__init__.py)
```

I'll see if I can work out a workaround